service/src/test/java/bio/terra/externalcreds/dataAccess/SshKeyPairInternalDaoTest.java

```diff
@@ -84,6 +121,88 @@ void testGetSshKeyPair() throws NoSuchAlgorithmException, IOException {
       assertPresent(loadedSshKeyOptional);
       verifySshKeyPair(sshKey, loadedSshKeyOptional.get());
     }
+
+    @Test
+    void testGetDecryptedKeyPair() throws NoSuchAlgorithmException, IOException {
+      var sshKey = createRandomGithubSshKey();
+      var cypheredkey = "jfidosruewr1k=";
+      when(externalCredsConfig.getEnableKmsEncryption()).thenReturn(true);
+      String location = "us-central1";
+      when(externalCredsConfig.getKeyRingLocation()).thenReturn(Optional.of(location));
+      String keyRing = "key_ring";
+      when(externalCredsConfig.getKeyRingId()).thenReturn(Optional.of(keyRing));
+      String encryptionKey = "encryption_key";
+      when(externalCredsConfig.getKeyId()).thenReturn(Optional.of(encryptionKey));
+      try (var utilsMock = Mockito.mockStatic(EncryptDecryptUtils.class)) {
+        utilsMock
+            .when(
+                () ->
+                    EncryptDecryptUtils.encryptSymmetric(
+                        eq(externalCredsConfig.getServiceGoogleProject()),
+                        eq(location),
+                        eq(keyRing),
+                        eq(encryptionKey),
+                        eq(sshKey.getPrivateKey())))
+            .thenReturn(cypheredkey);
+        utilsMock
+            .when(
+                () ->
+                    EncryptDecryptUtils.decryptSymmetric(
+                        eq(externalCredsConfig.getServiceGoogleProject()),
+                        eq(location),
+                        eq(keyRing),
+                        eq(encryptionKey),
+                        eq(cypheredkey)))
+            .thenReturn(sshKey.getPrivateKey());
+
+        sshKeyPairDAO.upsertSshKeyPair(sshKey);
+
+        var loadedSshKeyOptional =
+            sshKeyPairDAO.getSshKeyPair(sshKey.getUserId(), sshKey.getType());
```

Comment:
I think you should verify that the database contains the cyphertext. As the test is written, you can't be certain that the encryption was actually used.

Reply:
Also https://sonarcloud.io/project/issues?pullRequest=122&issues=AX_12dPla-RX1g1tBSPF&open=AX_12dPla-RX1g1tBSPF&id=DataBiosphere_terra-external-credentials-manager here

Reply:
i added a query to the database to check that but actually the mock should be able to check it as well as it should only return the de-cyphered text if the input matches the cyphered text.